packages/flame/lib/src/camera/camera_component.dart

```diff
@@ -299,13 +299,25 @@ class CameraComponent extends Component {
     viewfinder.add(MoveByEffect(offset, EffectController(speed: speed)));
   }
 
+  /// Allows to set the bounds of the camera component according to your world
+  /// component so that no empty space is displayed.
+  void adaptBoundsToWorld(SizeProvider world) {
+    final halfViewportSize = viewport.size / 2;
+    final worldSize = world.size; // Size from the center in each direction.
```

Comment:
The world doesn't have a size by default, I think it is better to provide an optional boolean to the `setBounds` method whether it should allow showing the black space outside of the world, not sure what a good name could be though.